test/com/google/enterprise/adaptor/filenet/PermissionsTest.java

```diff
@@ -691,4 +694,275 @@ public void testMarking_UserNotMatchingAnyAces() {
+
+    assertEquals(
+        ImmutableSet.<String>of("MARKING_allow_user_0", "MARKING_allow_user_1"),
+        acl.getAllowUsers());
+    assertEquals(
+        ImmutableSet.<String>of("MARKING_allow_group_0", AUTHENTICATED_USERS),
+        acl.getAllowGroups());
+    assertEquals(
+        ImmutableSet.<String>of("MARKING_deny_user_0", "MARKING_deny_user_1"),
+        acl.getDenyUsers());
+    assertEquals(
+        ImmutableSet.<String>of("MARKING_deny_group_0", "MARKING_deny_group_1"),
+        acl.getDenyGroups());
+  }
+
+  @SuppressWarnings("deprecation")  // For PermissionSource.MARKING
+  @Test
+  public void testMarkingAcl_allUseMarking_denyConstraintMask()
+      throws Exception {
+    Permissions testPerms = new Permissions(
+        TestObjectFactory.newPermissionList(
+            TestObjectFactory.generatePermissions(2, 2, 1, 2,
+                AccessRight.USE_MARKING_AS_INT, 0, PermissionSource.MARKING)));
+    Permissions.Acl acl = testPerms.getMarkingAcl(VIEW_ACCESS_RIGHTS);
+
+    assertEquals(
+        ImmutableSet.<String>of("MARKING_allow_user_0", "MARKING_allow_user_1"),
+        acl.getAllowUsers());
+    assertEquals(
+        ImmutableSet.<String>of("MARKING_allow_group_0"),
+        acl.getAllowGroups());
+    assertEquals(
+        ImmutableSet.<String>of("MARKING_deny_user_0", "MARKING_deny_user_1"),
+        acl.getDenyUsers());
+    assertEquals(
+        ImmutableSet.<String>of("MARKING_deny_group_0", "MARKING_deny_group_1"),
+        acl.getDenyGroups());
+  }
+
+  @SuppressWarnings("deprecation")  // For PermissionSource.MARKING
+  @Test
+  public void testMarkingAcl_allUseMarking_partialConstraintMask()
+      throws Exception {
+    Permissions testPerms = new Permissions(
+        TestObjectFactory.newPermissionList(
+            TestObjectFactory.generatePermissions(2, 2, 1, 2,
+                AccessRight.USE_MARKING_AS_INT, 0, PermissionSource.MARKING)));
+    Permissions.Acl acl = testPerms.getMarkingAcl(AccessRight.READ_AS_INT);
+
+    assertEquals(
+        ImmutableSet.<String>of("MARKING_allow_user_0", "MARKING_allow_user_1"),
+        acl.getAllowUsers());
+    assertEquals(
+        ImmutableSet.<String>of("MARKING_allow_group_0"),
+        acl.getAllowGroups());
+    assertEquals(
+        ImmutableSet.<String>of("MARKING_deny_user_0", "MARKING_deny_user_1"),
+        acl.getDenyUsers());
+    assertEquals(
+        ImmutableSet.<String>of("MARKING_deny_group_0", "MARKING_deny_group_1"),
+        acl.getDenyGroups());
+  }
+
+  @SuppressWarnings("deprecation")  // For PermissionSource.MARKING
+  @Test
+  public void testMarkingAcl_someUseMarking_allowConstraintMask()
+      throws Exception {
+    addAce(PermissionSource.MARKING, SecurityPrincipalType.USER,
+        AccessType.ALLOW, AccessRight.USE_MARKING_AS_INT, 0, "allowUser1");
+    addAce(PermissionSource.MARKING, SecurityPrincipalType.USER,
+        AccessType.ALLOW, 0, 0, "allowUser2");
```

Comment:
s/0/AccessRight.NONE_AS_INT/g-ish (as applicable). Also, why always 0 and not sometimes, say, VIEW_ACCESS_RIGHTS to mess with us.